### ISSUE TITLE
[MNG-3073] Memory Leak in maven2 for multiproject

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#4829]
**[Nicolas Cazottes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ncazottes)** opened **[MNG-3073](https://issues.apache.org/jira/browse/MNG-3073?redirect=false)** and commented

We use Maven2 for a massive multiproject composed of 433 jars projects in 3 levels of hierarchy and lots of inter-dependencies.
Our project contains 3577 classes at this time but it is constantly growing.

As it grows, the number of projects increases and the number of versions of each project increases as well.

What we are faced to is that we constantly need to increase the memory size allowed to Maven.
With 433 projects and 3577 classes, we need the following memory parameters : -Xmx900m -Xms128m for our compilation with the following command line : mvn install -Dmaven.test.skip=true.

I think there are memory leak in Maven that explains this behaviour.

---

**Affects:** 2.0.7

3 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Debabrat Panda](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=debabratp)** commented

Set the memory size inside "Maven2.0.4/bin/mvn.bat" file
like this set ANT_OPTS= -Xmx1024m

for your reference where to set ANT_OPTS
`@REM` Reaching here means variables are defined and arguments have been captured
:endInit
SET MAVEN_JAVA_EXE="%JAVA_HOME%\bin\java.exe"
set ANT_OPTS= -Xmx1024m

**Comment by jira-importer** [comment]
**[Milos Kleint](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mkleint)** commented

forking the compiler could have influence on memory usage as well. Haven't measured rigorously though.

**Comment by jira-importer** [comment]
**[Nicolas Cazottes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ncazottes)** commented

I actually set the memory configuration I mentionned through the environment variable MAVEN_OPTS.

My point is that the solution to increase the memory endless is not a good one because I may be at some time faced to a limit of my computer memory size if my number of projects increases to 1000.

Forking the compiler should be an option but I think it will increase the compilation time which is currently 12min average. I will try it to see.
